src/main.rs

```diff
@@ -1,25 +1,34 @@
-extern crate requests;
-extern crate bigdecimal;
-extern crate json;
+extern crate reqwest;
+extern crate rust_decimal;
+extern crate serde;
 
-// adds method .json() to the Response class
-use requests::ToJson;
+use rust_decimal::Decimal;
+use serde::Deserialize;
 
-use std::str::FromStr;
-use bigdecimal::BigDecimal;
+#[derive(Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+struct Rates {
+    // btc: Decimal,
+    eth: Decimal,
+}
 
-//use requests::{Codes, Request, Response, StatusCode};
+#[derive(Deserialize)]
+struct Data {
+    // currency: String,
+    rates: Rates,
+}
 
-fn main() {
+#[derive(Deserialize)]
+struct Body {
+    data: Data,
+}
 
+fn main() -> reqwest::Result<()> {
```

Comment:
It's necessary to have this return type and to explicitly pass a Response so that we can use the `?` operator to propagate exceptions out of our scope. It would probably be better to match on the Result objects, since it doesn't make much sense for a main function to have a return.